### PR TITLE
fix(sync): reduce the memory usage in synchronization

### PR DESCRIPTION
**Pull request by KaoImin**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Which docs this PR relation**:

Ref #

**Which toolchain this PR adaption**:

No Breaking Change

**Special notes for your reviewer**:

### CODE REVIEW

File: core/synchronization/src/task.rs [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    );
+
+    commit_transaction(tx).await?;
+
+    Ok(())
+}
+
+pub async fn sync_indexer_cells(sub_task: &[u64], rdb: XSQLPool) -> Result<()> {
+    let mut indexer_cells = Vec::new();
+    // let mut status_list = Vec::new();
```
Comment (quake) [inline core/synchronization/src/task.rs:234]: please remove commented out code

File: core/synchronization/src/task.rs [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+                    cell.input_index.unwrap(),
+                    cell.consumed_tx_hash.clone(),
+                    cell.consumed_tx_index.unwrap(),
+                );
+                indexer_cells.push(i_cell.update_by_cell_table(cell));
+            }
+        }
+    }
+
+    // status_list.extend(sub_task.iter().map(|num| SyncStatus::new(*num)));
```
Comment (quake) [inline core/synchronization/src/task.rs:270]: please remove commented out code

**Review by quake — CHANGES_REQUESTED** [review]

File: core/cli/src/lib.rs [hunk 3]
```diff
@@ -142,10 +142,7 @@ impl<'a> Cli<'a> {
 
         if self.config.sync_mode && self.config.allow_parallel_sync {
             service
-                .do_sync(
-                    self.config.sync_config.sync_block_batch_size,
-                    self.config.sync_config.max_task_number,
-                )
+                .do_sync(self.config.sync_config.sync_block_batch_size)
```
Comment (EthanYuan) [inline core/cli/src/lib.rs:145]: The do_sync method accepts a max_task_number parameter, but passes the sync_block_batch_size parameter.

**Review by quake — APPROVED** [review]

File: core/synchronization/src/lib.rs [hunk 4]
```diff
@@ -118,171 +90,105 @@ impl<T: SyncAdapter> Synchronization<T> {
...
-        });
-
-        log::info!("[sync] to sync indexer cell {}", to_sync_number_set.len());
-        Ok(to_sync_number_set.into_iter().collect())
-    }
-
-    async fn get_sync_indexer_completed_numbers(&self) -> Result<Vec<BlockNumber>> {
-        let mut conn = self.pool.acquire().await?;
-        let res = conn.fetch_list::<SyncStatus>().await?;
+        for id in (0..self.chain_tip).step_by(TASK_LEN as usize) {
```
Comment (quake) [inline core/synchronization/src/lib.rs:147]: should be `=self.chain_tip` ?
```suggestion
        for id in (0..=self.chain_tip).step_by(TASK_LEN as usize) {
```

**Review by EthanYuan — APPROVED** [review]

**Pull request merged**